storage/gcp/gcp.go

```diff
@@ -949,6 +1008,58 @@ func (s *spannerCoordinator) publishTree(ctx context.Context, minAge time.Durati
 	return nil
 }
 
+// garbageCollect is a long running function which will identify unneeded partial tiles/entry bundles, and call the provided function to remove them.
+//
+// Uses the `GCCoord` table to ensure that only one binary is actively garbage collecting at any given time, and to track progress so that we don't
+// needlessly attempt to GC over regions which have already been cleaned.
+//
+// Returns true if we've "caught up" with the current state of the tree.
+func (s *spannerCoordinator) garbageCollect(ctx context.Context, treeSize uint64, maxDeletes uint, deleteWithPrefix func(ctx context.Context, prefix string) error) error {
```

Comment:
I discuss this in the comments below, but `maxDeletes` required me to update my intuition. In addition to the non-strict nature of it, my expectations about what it counted as a deletion were wrong.

What I _expected_ was that this would control the maximum number of partial tiles that would be deleted. It actually controls the number of calls to `deleteWithPrefix`, each of which could delete up to 255 partial tiles. Nothing wrong with this, but worth having a doc line to make it clear.